Maple2.Server.Game/Manager/ShopManager.cs

```diff
@@ -85,162 +85,211 @@ private bool TryGetShopData(int shopId, [NotNullWhen(true)] out CharacterShopDat
         return accountShopData.TryGetValue(shopId, out data) || characterShopData.TryGetValue(shopId, out data);
     }
 
+    private bool TryRemoveShopData(int shopId) {
+        if (accountShopData.Remove(shopId)) {
+            return true;
+        }
+        return characterShopData.Remove(shopId);
+    }
+
     private bool TryGetShopItemData(int shopId, [NotNullWhen(true)] out IDictionary<int, CharacterShopItemData>? data) {
         return accountShopItemData.TryGetValue(shopId, out data) || characterShopItemData.TryGetValue(shopId, out data);
     }
 
+    /// <summary>
+    /// Load shop.
+    /// </summary>
+    /// <param name="shopId">Shop ID</param>
+    /// <param name="npcId">Npc ID. Optional parameter. This only is used for displaying the NPC's name on the shop window top bar.</param>
     public void Load(int shopId, int npcId = 0) {
-        Shop? shop = session.FindShop(shopId);
-        if (shop == null) {
-            logger.Warning("Shop {Id} has not been implemented", shopId);
+        if (!session.ServerTableMetadata.ShopTable.Entries.TryGetValue(shopId, out ShopMetadata? metadata)) {
```

Comment:
**suggestion (performance):** New data-driven approach using ServerTableMetadata

This change moves towards a more configurable system. Consider adding a comment explaining the benefits of this new approach and any potential performance considerations.